File: cli/und.lua
#!/bin/env luajit

local args = {...}
local target = args[1]
local command = args[2]

if (not command) then
	command = "help"
elseif (command == "und") then
	command = "help"
end

local commandMap = {
	["-h"] = "help",
	["-v"] = "version"
}

if (commandMap[command]) then
	command = commandMap[command]
end

-- Hooray for globals!
CLI_COMMAND = command
CLI_TARGET = target

local result, err = pcall(require, "cli." .. command)
if (not result) then
	print(err)
	print("Unknown und command '" .. args[2] .. "'")
	os.exit(1)
end

File: bin/und.js
#!/usr/bin/env node
"use strict";

const commandMap = {
	"-h": "help",
	"-v": "version"
};

let args = process.argv;
let cmd = args[2];

if (!cmd) {
	cmd = "help";
}

if (commandMap[cmd]) {
	cmd = commandMap[cmd];
}

try {
	require("../cli/" + cmd);
} catch(e) {
	if (e.code === "MODULE_NOT_FOUND") {
		console.error(`Unknown command "${cmd}"; try "und help"`);
		return;
	}

	throw e;
}